sidecar/internal/service/reclaimspace.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2021 The Kubernetes-CSI-Addons Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package service
+
+import (
+	"context"
+
+	"github.com/csi-addons/kubernetes-csi-addons/internal/proto"
+	csiReclaimSpace "github.com/csi-addons/spec/lib/go/reclaimspace"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"google.golang.org/grpc"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+// ReclaimSpaceServer struct of sidecar with supported methods of proto
+// reclaim space server spec and also containing reclaimspace node and
+// controller client to csi driver.
+type ReclaimSpaceServer struct {
+	proto.UnimplementedReclaimSpaceServer
+	controllerClient csiReclaimSpace.ReclaimSpaceControllerClient
+	nodeClient       csiReclaimSpace.ReclaimSpaceNodeClient
+	kubeClient       *kubernetes.Clientset
```

Comment:
why this is a pointer to kubernetes clientset?

Reply:
> why this is a pointer to kubernetes clientset?

All the other services will share a kubeClient(passed in main.go), I don't see a reason to make a copy each time.